sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager_base.py

```diff
@@ -25,24 +26,106 @@
+        seed=123abc\ndefault_when_enabled=Control\npercentiles=0,Control,20;20,Test,100\nvariants=Control,standard;Test,special # pylint:disable=line-too-long
+
+        :param  Dict[str, JSON] feature_flag_value: The feature to generate an allocation ID for.
+        :rtype: str
+        :return: The allocation ID.
+        """
+
+        allocation_id = ""
+        allocated_variants = []
+
+        allocation: Optional[JSON] = feature_flag_value.get("allocation")
+
+        if allocation:
+            # Seed
+            allocation_id = f"seed={allocation.get('seed', '')}"
+
+            # DefaultWhenEnabled
+            if "default_when_enabled" in allocation:
+                allocated_variants.append(allocation.get("default_when_enabled"))
+
+            allocation_id += f"\ndefault_when_enabled={allocation.get('default_when_enabled', '')}"
+
+            # Percentile
+            allocation_id += "\npercentiles="
+
+            percentile = allocation.get("percentile")
+
+            if percentile:
+                percentile_allocations = sorted(
+                    (x for x in percentile if x.get("from") != x.get("to")),
+                    key=lambda x: x.get("from"),
+                )
+
+                for percentile_allocation in percentile_allocations:
+                    if "variant" in percentile_allocation:
+                        allocated_variants.append(percentile_allocation.get("variant"))
+
+                allocation_id += ";".join(
+                    f"{pa.get('from')},"
+                    f"{base64.b64encode(pa.get('variant').encode('utf-8')).decode('utf-8')},"
+                    f"{pa.get('to')}"
+                    for pa in percentile_allocations
+                )
+        else:
+            allocation_id = "seed=\ndefault_when_enabled=\npercentiles="
+
+        # Variants
+        allocation_id += "\nvariants="
+
+        variants_value = feature_flag_value.get("variants")
+        if variants_value and (isinstance(variants_value, list) or all(isinstance(v, dict) for v in variants_value)):
+            if allocated_variants:
+                if isinstance(variants_value, list) and all(isinstance(v, dict) for v in variants_value):
+                    sorted_variants: List[Dict[str, Any]] = sorted(
+                        (v for v in variants_value if v.get("name") in allocated_variants),
+                        key=lambda v: v.get("name"),
+                    )
+
+                    for v in sorted_variants:
+                        allocation_id += f"{base64.b64encode(v.get('name', '').encode('utf-8')).decode('utf-8')},"
+                        if "configuration_value" in v:
+                            allocation_id += f"{json.dumps(v.get('configuration_value', ''), separators=(',', ':'))}"
+                        allocation_id += ";"
+                    allocation_id = allocation_id[:-1]
+
+        if not allocated_variants and (not allocation or not allocation.get("seed")):
```

Comment:
This check can be done before variants (since we're no longer adding to the `allocated_variants`)

Reply:
My question was on the phrasing of `(since we're no longer adding to the allocated_variants)`. It makes it sound like we should never do the allocations check.

Reply:
No- I just meant within this code from top to bottom. We check allocations and add variants that are actually allocated to the allocated_variants. Then halfway through this code, we're done and no longer adding to this array.

Reply:
I've already moved it. Is it good in it's new location?